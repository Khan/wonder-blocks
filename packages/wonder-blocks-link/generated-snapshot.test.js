// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-link
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");
import Link from "./components/link.js";

describe("wonder-blocks-link", () => {
    it("example 1", () => {
        const Color = require("@khanacademy/wonder-blocks-color").default;
        const {View} = require("@khanacademy/wonder-blocks-core");

        const example = (
            <View>
                <p>
                    I am a <Link href="#nonexistent-link">Primary Link</Link>.{" "}
                    <span style={{color: Color.offBlack64}}>
                        My friend the
                        <Link
                            href="#secondary-nonexistent-link"
                            kind="secondary"
                        >
                            Secondary Link
                        </Link>{" "}
                        is used here with a lighter text.
                    </span>{" "}
                    We also have a
                    <Link href="#" visitable={true}>
                        Visitable Primary Link
                    </Link>{" "}
                    friend.
                </p>
                <p
                    style={{
                        backgroundColor: Color.darkBlue,
                        color: Color.white64,
                        padding: 10,
                    }}
                >
                    I am a{" "}
                    <Link href="#dark-link" light={true}>
                        Primary Link
                    </Link>{" "}
                    used on a dark background. My friend the Secondary Link
                    isn't supported on this dark background.
                </p>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const {HeadingSmall} = require("@khanacademy/wonder-blocks-typography");
        const {View} = require("@khanacademy/wonder-blocks-core");

        const example = (
            <View>
                <Link href="#nonexistent-link">
                    <HeadingSmall>Heading inside a Link element</HeadingSmall>
                </Link>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
