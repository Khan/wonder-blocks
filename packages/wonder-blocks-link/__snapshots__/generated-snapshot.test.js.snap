// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-link example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <p>
    I am a 
    <a
      className=""
      href="#nonexistent-link"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "color": "#1865f2",
          "cursor": "pointer",
          "outline": "none",
          "textDecoration": "none",
        }
      }
      tabIndex={0}
    >
      Primary Link
    </a>
    .
     
    <span
      style={
        Object {
          "color": "rgba(33,36,44,0.64)",
        }
      }
    >
      My friend the
      <a
        className=""
        href="#secondary-nonexistent-link"
        onBlur={[Function]}
        onClick={[Function]}
        onDragStart={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "color": "#21242c",
            "cursor": "pointer",
            "outline": "none",
            "textDecoration": "none",
          }
        }
        tabIndex={0}
      >
        Secondary Link
      </a>
       
      is used here with a lighter text.
    </span>
     
    We also have a
    <a
      className=""
      href="#"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          ":visited": Object {
            "color": "#8755ee",
          },
          "color": "#1865f2",
          "cursor": "pointer",
          "outline": "none",
          "textDecoration": "none",
        }
      }
      tabIndex={0}
    >
      Visitable Primary Link
    </a>
     
    friend.
  </p>
  <p
    style={
      Object {
        "backgroundColor": "#0a2a66",
        "color": "rgba(255,255,255,0.64)",
        "padding": 10,
      }
    }
  >
    I am a
     
    <a
      className=""
      href="#dark-link"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "color": "#ffffff",
          "cursor": "pointer",
          "outline": "none",
          "textDecoration": "none",
        }
      }
      tabIndex={0}
    >
      Primary Link
    </a>
     
    used on a dark background. My friend the Secondary Link isn't supported on this dark background.
  </p>
</div>
`;

exports[`wonder-blocks-link example 2 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <a
    className=""
    href="#nonexistent-link"
    onBlur={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "color": "#1865f2",
        "cursor": "pointer",
        "outline": "none",
        "textDecoration": "none",
      }
    }
    tabIndex={0}
  >
    <h4
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 20,
          "fontWeight": 700,
          "lineHeight": "24px",
          "marginBottom": 0,
          "marginTop": 0,
        }
      }
    >
      Heading inside a Link element
    </h4>
  </a>
</div>
`;
